File: worblehat-web/src/main/java/de/codecentric/psd/Worblehat.java
package de.codecentric.psd;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.ConfigurableApplicationContext;

import java.util.Scanner;

@SpringBootApplication
public class Worblehat {

	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		ConfigurableApplicationContext applicationContext = SpringApplication.run(Worblehat.class, args);

		String line;
		do {
			line = scan.nextLine();
		} while (!"stop".equals(line));
		applicationContext.close();
	}
}

File: worblehat-web/src/main/java/de/codecentric/psd/Worblehat.kt
package de.codecentric.psd

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import java.util.*

@SpringBootApplication
open class Worblehat {

    companion object {
        @JvmStatic fun main(args: Array<String>) {
            Worblehat().run(args)
        }
    }

    fun run(args: Array<String>) {
        val scan = Scanner(System.`in`)
        val applicationContext = SpringApplication.run(Worblehat::class.java, *args)

        var line: String
        do {
            line = scan.nextLine()
        } while ("stop" != line)
        applicationContext.close()
    }
}
